src/app/admin/pages/users/users.component.html

```diff
@@ -29,12 +29,15 @@
 
   <div class="users" main>
     <div *ngIf="!loading; else loadingTemplate">
-      <div *ngIf="users?.length; else emptyTemplate" class="user-cards">
-        <clark-admin-user-card *ngFor="let user of users" class="user-cards__card"
+      <div *ngIf="users?.length; else emptyTemplate" class="user-cards" [@staggerChildren]>
+        <clark-admin-user-card *ngFor="let user of users;trackBy: trackby" class="user-cards__card"
```

Comment:
Can you explain this change? Not sure what this does.

Reply:
The`trackBy` function gives the `*ngFor` directive the ability to track changes to the list it represents. Without it, if I give `*ngFor` an array of learning objects and remove one of them, it doesn't know which one I removed so it re-renders the entire list. The `trackBy` function in the component Typescript takes an index and an objects and returns the property by which to track, in this case the learning object id. So now, if I add or remove an object from the list the directive knows what was changed and can re-render only that piece. More here https://netbasal.com/angular-2-improve-performance-with-trackby-cc147b5104e5